### DISCUSSION TITLE
Add realtime filter function.

### DISCUSSION THREAD

**Discussion by morgan55555** [open-webui/open-webui#7223] [Ideas]
Currently, there is only inlet and outlet in filters.
It means that if you want to dynamically format LLM's output, you should write own pipe, in which you need to implement all of open webui functionality, just for simple output formatting task.

E.g., you want to make model, which will reflect and then you want to hide model reflection.
And currently you need to make this with pipe functionality (and implement much of webui already implemented functions), or use outlet (and this filter can only be performed after all generation is completed).

There will be great to use some middle filter, like outlet, but called on the fly, so we can use it functionality to modify any output dynamically, just after it showed on the screen.

**Comment by jcarlosroldan** [comment]
I've been a few hours trying to do this! Some way of having not just `Filter.inlet` and `.outlet` but also `.midlet` (if that's even a word). If this feature is interesting I'd be happy to contribute to Open WebUI to add it @tjbck

**Reply by ddddavid-he** [reply]
There is a `stream` available in `Filter` class to due with stream contents in the newer versions according to the document. 
[https://docs.openwebui.com/features/plugin/functions/filter](https://docs.openwebui.com/features/plugin/functions/filter)

You can do things like this:

```python
class Filter:
    ...

    def stream(self, event: dict) -> dict:
        ...
        return event

    ...

```
